lib/puppet/util/windows/registry.rb

```diff
@@ -31,16 +31,24 @@ def open(name, path, mode = KEY_READ | KEY64, &block)
     end
 
     def values(subkey)
+      orig_enc = Encoding.default_internal
+      Encoding.default_internal = Encoding::UTF_8
```

Comment:
Sorry but I'm not following why this is necessary. Ruby uses the wide versions to read data from the registry. Strings should have UTF-16LE encoding at that point. When [ruby's encode method](https://github.com/ruby/ruby/blob/v2_1_5/ext/win32/lib/win32/registry.rb#L894-L896) is called, it should convert from UTF-16LE to the native codepage encoding, e.g. CP1252, and yield to us a CP1252 encoded string.

<pre>
subkey.each_value do |name, type, data|
  # if type == REG_SZ, then data should be CP1252 encoded
end
</pre>

We should then be able to convert that string to UTF-8, `data.encode('UTF-8')`, and we should be able to skip the non-sense that we were doing earlier in https://github.com/ferventcoder/puppet/blob/ca76c70d06bcbb379b28f4549a38f66a43176623/lib/puppet/util/windows/registry.rb#L55-L78